src/js_printer.zig

```diff
@@ -489,28 +483,14 @@ pub const RequireOrImportMeta = struct {
 };
 
 pub const PrintResult = union(enum) {
-    result: struct {
-        code: []u8,
-        source_map: ?SourceMap.Chunk = null,
-    },
+    result: Success,
     err: anyerror,
 
-    pub fn clone(
-        this: PrintResult,
-        allocator: std.mem.Allocator,
-    ) !PrintResult {
-        return switch (this) {
-            .result => PrintResult{
-                .result = .{
-                    .code = try allocator.dupe(u8, this.result.code),
-                    .source_map = this.result.source_map,
-                },
-            },
-            .err => PrintResult{
-                .err = this.err,
-            },
-        };
-    }
+    pub const Success = struct {
+        code: []u8,
+        code_allocator: ?std.mem.Allocator,
```

Comment:
Since this is set to null only when `code` is empty, I think we can change this to a non-optional `std.mem.Allocator`, and just use `bun.default_allocator` instead of null. This will save space since Zig won't optimize the optional